Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/mysql/MySQLMvpModel.java

```diff
@@ -0,0 +1,301 @@
+        resultList.add(ServerVersion.FIVE_FULL_STOP_SEVEN.toString());
+        resultList.add(ServerVersion.EIGHT_FULL_STOP_ZERO.toString());
+        return resultList;
+    }
+
+    public static List<Region> listSupportedRegions() {
+        return Arrays.asList(com.microsoft.azure.arm.resources.Region.values()).stream()
+                .filter(e -> MYSQL_SUPPORTED_REGIONS.contains(e.name()))
+                .map(e -> Region.findByLabelOrName(e.name()))
+                .sorted(Comparator.comparing(Region::label))
+                .collect(Collectors.toList());
+    }
+
+    public static boolean updateSSLEnforcement(final String subscriptionId, final Server server, final SslEnforcementEnum sslEnforcement) {
+        final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        final ServerUpdateParameters parameters = new ServerUpdateParameters();
+        parameters.withSslEnforcement(sslEnforcement);
+        mySQLManager.servers().inner().update(server.resourceGroupName(), server.name(), parameters);
+        return true;
+    }
+
+    public static class DatabaseMvpModel {
+
+        public static List<DatabaseInner> listDatabases(final String subscriptionId, final Server server) {
+            final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+            final List<DatabaseInner> databases = mySQLManager.databases().inner().listByServer(server.resourceGroupName(), server.name());
+            return databases;
+        }
+    }
+
+    public static class FirewallRuleMvpModel {
+
+        public static List<FirewallRuleInner> listFirewallRules(final String subscriptionId, final Server server) {
+            final MySQLManager manager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+            final List<FirewallRuleInner> firewallRules = manager.firewallRules().inner().listByServer(server.resourceGroupName(), server.name());
+            return firewallRules;
+        }
+
+        public static List<FirewallRuleInner> listFirewallRules(final String subscriptionId, final String resourceGroupName, final String name) {
+            final MySQLManager manager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+            final List<FirewallRuleInner> firewallRules = manager.firewallRules().inner().listByServer(resourceGroupName, name);
+            return firewallRules;
+        }
+
+        public static boolean isAllowAccessFromAzureServices(final String subscriptionId, final Server server) {
+            List<FirewallRuleInner> firewallRules = MySQLMvpModel.FirewallRuleMvpModel.listFirewallRules(subscriptionId, server);
+            return MySQLMvpModel.FirewallRuleMvpModel.isAllowAccessFromAzureServices(firewallRules);
+        }
+
+        public static boolean isAllowAccessFromAzureServices(final List<FirewallRuleInner> firewallRules) {
```

Comment:
May enhance the validation later as users may rename or change the firewall rules

Reply:
no. the name of this rule is fixed.

Reply:
Could users modified rule in portal or cli tools?